Compiler warnings when building Oboe for CTS Verifier

make -j CtsVerifier
in Android results in these compiler warnings.


external/oboe/src/common/AudioStreamBuilder.cpp:99:50: warning: format specifies type 'int' but the argument has type 'Result' [-Wformat]
   99 |         LOGW("%s() invalid config %d", __func__, result);
      |                                   ~~             ^~~~~~
      |                                                  static_cast<int>(
external/oboe/src/common/OboeDebug.h:29:70: note: expanded from macro 'LOGW'
   29 | #define LOGW(...) __android_log_print(ANDROID_LOG_WARN, MODULE_NAME, __VA_ARGS__)
      |                                                                      ^~~~~~~~~~~
1 warning generated.
[ 90% 924/1020] //external/oboe:oboe clang++ src/opensles/AudioOutputStreamOpenSLES.cpp
external/oboe/src/opensles/AudioOutputStreamOpenSLES.cpp:375:70: warning: format specifies type 'int' but the argument has type 'Result' [-Wformat]
  375 |             LOGW("Failed to clear buffer queue. OpenSLES error: %d", result);
      |                                                                 ~~   ^~~~~~
      |                                                                      static_cast<int>(
external/oboe/src/common/OboeDebug.h:29:70: note: expanded from macro 'LOGW'
   29 | #define LOGW(...) __android_log_print(ANDROID_LOG_WARN, MODULE_NAME, __VA_ARGS__)
      |                                                                      ^~~~~~~~~~~
1 warning generated.
[ 90% 926/1020] //external/oboe:oboe clang++ src/aaudio/AudioStreamAAudio.cpp
external/oboe/src/aaudio/AudioStreamAAudio.cpp:531:66: warning: format specifies type 'int' but the argument has type 'DataCallbackResult' [-Wformat]
  531 |             LOGE("Oboe callback returned unexpected value = %d", result);
      |                                                             ~~   ^~~~~~
      |                                                                  static_cast<int>(
external/oboe/src/common/OboeDebug.h:30:71: note: expanded from macro 'LOGE'
   30 | #define LOGE(...) __android_log_print(ANDROID_LOG_ERROR, MODULE_NAME, __VA_ARGS__)
      |                                                                       ^~~~~~~~~~~
1 warning generated.
[ 91% 933/1020] //external/oboe:oboe clang++ src/common/DataConversionFlowGraph.cpp
external/oboe/src/common/DataConversionFlowGraph.cpp:99:13: warning: format specifies type 'int' but the argument has type 'AudioFormat' [-Wformat]
   95 |     LOGI("%s() flowgraph converts channels: %d to %d, format: %d to %d"
      |                                                               ~~
   96 |          ", rate: %d to %d, cbsize: %d to %d, qual = %d",
   97 |             __func__,
   98 |             sourceChannelCount, sinkChannelCount,
   99 |             sourceFormat, sinkFormat,
      |             ^~~~~~~~~~~~
      |             static_cast<int>(
external/oboe/src/common/OboeDebug.h:28:70: note: expanded from macro 'LOGI'
   28 | #define LOGI(...) __android_log_print(ANDROID_LOG_INFO, MODULE_NAME, __VA_ARGS__)
      |                                                                      ^~~~~~~~~~~
external/oboe/src/common/DataConversionFlowGraph.cpp:99:27: warning: format specifies type 'int' but the argument has type 'AudioFormat' [-Wformat]
   95 |     LOGI("%s() flowgraph converts channels: %d to %d, format: %d to %d"
      |                                                                     ~~
   96 |          ", rate: %d to %d, cbsize: %d to %d, qual = %d",
   97 |             __func__,
   98 |             sourceChannelCount, sinkChannelCount,
   99 |             sourceFormat, sinkFormat,
      |                           ^~~~~~~~~~
      |                           static_cast<int>(
external/oboe/src/common/OboeDebug.h:28:70: note: expanded from macro 'LOGI'
   28 | #define LOGI(...) __android_log_print(ANDROID_LOG_INFO, MODULE_NAME, __VA_ARGS__)
      |                                                                      ^~~~~~~~~~~
external/oboe/src/common/DataConversionFlowGraph.cpp:102:13: warning: format specifies type 'int' but the argument has type 'SampleRateConversionQuality' [-Wformat]
   96 |          ", rate: %d to %d, cbsize: %d to %d, qual = %d",
      |                                                      ~~
   97 |             __func__,
   98 |             sourceChannelCount, sinkChannelCount,
   99 |             sourceFormat, sinkFormat,
  100 |             sourceSampleRate, sinkSampleRate,
  101 |             sourceFramesPerCallback, sinkFramesPerCallback,
  102 |             sourceStream->getSampleRateConversionQuality());
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |             static_cast<int>(
external/oboe/src/common/OboeDebug.h:28:70: note: expanded from macro 'LOGI'
   28 | #define LOGI(...) __android_log_print(ANDROID_LOG_INFO, MODULE_NAME, __VA_ARGS__)
      |                                                                      ^~~~~~~~~~~
external/oboe/src/common/DataConversionFlowGraph.cpp:131:78: warning: format specifies type 'int' but the argument has type 'AudioFormat' [-Wformat]
  131 |                 LOGE("%s() Unsupported source caller format = %d", __func__, sourceFormat);
      |                                                               ~~             ^~~~~~~~~~~~
      |                                                                              static_cast<int>(
external/oboe/src/common/OboeDebug.h:30:71: note: expanded from macro 'LOGE'
   30 | #define LOGE(...) __android_log_print(ANDROID_LOG_ERROR, MODULE_NAME, __VA_ARGS__)
      |                                                                       ^~~~~~~~~~~
external/oboe/src/common/DataConversionFlowGraph.cpp:153:71: warning: format specifies type 'int' but the argument has type 'AudioFormat' [-Wformat]
  153 |                 LOGE("%s() Unsupported source format = %d", __func__, sourceFormat);
      |                                                        ~~             ^~~~~~~~~~~~
      |                                                                       static_cast<int>(
external/oboe/src/common/OboeDebug.h:30:71: note: expanded from macro 'LOGE'
   30 | #define LOGE(...) __android_log_print(ANDROID_LOG_ERROR, MODULE_NAME, __VA_ARGS__)
      |                                                                       ^~~~~~~~~~~
external/oboe/src/common/DataConversionFlowGraph.cpp:229:65: warning: format specifies type 'int' but the argument has type 'AudioFormat' [-Wformat]
  229 |             LOGE("%s() Unsupported sink format = %d", __func__, sinkFormat);
      |                                                  ~~             ^~~~~~~~~~
      |                                                                 static_cast<int>(
external/oboe/src/common/OboeDebug.h:30:71: note: expanded from macro 'LOGE'
   30 | #define LOGE(...) __android_log_print(ANDROID_LOG_ERROR, MODULE_NAME, __VA_ARGS__)
      |                                                                       ^~~~~~~~~~~
6 warnings generated.
[ 91% 934/1020] //external/oboe:oboe clang++ src/opensles/AudioStreamOpenSLES.cpp
external/oboe/src/opensles/AudioStreamOpenSLES.cpp:76:24: warning: format specifies type 'int' but the argument has type 'AudioFormat' [-Wformat]
   75 |         LOGW("%s() Android's OpenSL ES implementation only supports I16 and Float. Format: %d",
      |                                                                                            ~~
   76 |              __func__, mFormat);
      |                        ^~~~~~~
      |                        static_cast<int>(
external/oboe/src/common/OboeDebug.h:29:70: note: expanded from macro 'LOGW'
   29 | #define LOGW(...) __android_log_print(ANDROID_LOG_WARN, MODULE_NAME, __VA_ARGS__)
      |                                                                      ^~~~~~~~~~~
external/oboe/src/opensles/AudioStreamOpenSLES.cpp:425:62: warning: format specifies type 'int' but the argument has type 'DataCallbackResult' [-Wformat]
  425 |         LOGW("Oboe callback returned unexpected value = %d", result);
      |                                                         ~~   ^~~~~~
      |                                                              static_cast<int>(
external/oboe/src/common/OboeDebug.h:29:70: note: expanded from macro 'LOGW'
   29 | #define LOGW(...) __android_log_print(ANDROID_LOG_WARN, MODULE_NAME, __VA_ARGS__)

--- a/src/aaudio/AudioStreamAAudio.cpp
+++ b/src/aaudio/AudioStreamAAudio.cpp
@@ -528,7 +528,7 @@ DataCallbackResult AudioStreamAAudio::callOnAudioReady(AAudioStream * /*stream*/
         if (result == DataCallbackResult::Stop) {
             LOGD("Oboe callback returned DataCallbackResult::Stop");
         } else {
-            LOGE("Oboe callback returned unexpected value. Error: %s", oboe::convertToText(result));
+            LOGE("Oboe callback returned unexpected value. Error: %d", static_cast<int>(result));
         }
 
         // Returning Stop caused various problems before S. See #1230

--- a/src/aaudio/AudioStreamAAudio.cpp
+++ b/src/aaudio/AudioStreamAAudio.cpp
@@ -528,7 +528,7 @@ DataCallbackResult AudioStreamAAudio::callOnAudioReady(AAudioStream * /*stream*/
         if (result == DataCallbackResult::Stop) {
             LOGD("Oboe callback returned DataCallbackResult::Stop");
         } else {
-            LOGE("Oboe callback returned unexpected value = %d", result);
+            LOGE("Oboe callback returned unexpected value. Error: %s", oboe::convertToText(result));
         }
 
         // Returning Stop caused various problems before S. See #1230

--- a/src/common/AudioStreamBuilder.cpp
+++ b/src/common/AudioStreamBuilder.cpp
@@ -96,7 +96,7 @@ Result AudioStreamBuilder::openStream(AudioStream **streamPP) {
 Result AudioStreamBuilder::openStreamInternal(AudioStream **streamPP) {
     auto result = isValidConfig();
     if (result != Result::OK) {
-        LOGW("%s() invalid config %d", __func__, result);
+        LOGW("%s() invalid config. Error %s", __func__, oboe::convertToText(result));
         return result;
     }
 

--- a/src/common/DataConversionFlowGraph.cpp
+++ b/src/common/DataConversionFlowGraph.cpp
@@ -92,14 +92,14 @@ Result DataConversionFlowGraph::configure(AudioStream *sourceStream, AudioStream
     int32_t sinkSampleRate = sinkStream->getSampleRate();
     int32_t sinkFramesPerCallback = sinkStream->getFramesPerDataCallback();
 
-    LOGI("%s() flowgraph converts channels: %d to %d, format: %d to %d"
-         ", rate: %d to %d, cbsize: %d to %d, qual = %d",
+    LOGI("%s() flowgraph converts channels: %d to %d, format: %s to %s"
+            ", rate: %d to %d, cbsize: %d to %d, qual = %s",
             __func__,
             sourceChannelCount, sinkChannelCount,
-            sourceFormat, sinkFormat,
+            oboe::convertToText(sourceFormat), oboe::convertToText(sinkFormat),
             sourceSampleRate, sinkSampleRate,
             sourceFramesPerCallback, sinkFramesPerCallback,
-            sourceStream->getSampleRateConversionQuality());
+            oboe::convertToText(sourceStream->getSampleRateConversionQuality()));
 
     // Source
     // IF OUTPUT and using a callback then call back to the app using a SourceCaller.
@@ -128,7 +128,7 @@ Result DataConversionFlowGraph::configure(AudioStream *sourceStream, AudioStream
                                                                   actualSourceFramesPerCallback);
                 break;
             default:
-                LOGE("%s() Unsupported source caller format = %d", __func__, sourceFormat);
+                LOGE("%s() Unsupported source caller format = %s", __func__, oboe::convertToText(sourceFormat));
                 return Result::ErrorIllegalArgument;
         }
         mSourceCaller->setStream(sourceStream);
@@ -150,7 +150,7 @@ Result DataConversionFlowGraph::configure(AudioStream *sourceStream, AudioStream
                 mSource = std::make_unique<SourceI32>(sourceChannelCount);
                 break;
             default:
-                LOGE("%s() Unsupported source format = %d", __func__, sourceFormat);
+                LOGE("%s() Unsupported source format = %s", __func__, oboe::convertToText(sourceFormat));
                 return Result::ErrorIllegalArgument;
         }
         if (isInput) {
@@ -226,7 +226,7 @@ Result DataConversionFlowGraph::configure(AudioStream *sourceStream, AudioStream
             mSink = std::make_unique<SinkI32>(sinkChannelCount);
             break;
         default:
-            LOGE("%s() Unsupported sink format = %d", __func__, sinkFormat);
+            LOGE("%s() Unsupported sink format = %s", __func__, oboe::convertToText(sinkFormat));
             return Result::ErrorIllegalArgument;;
     }
     lastOutput->connect(&mSink->input);

--- a/src/common/DataConversionFlowGraph.cpp
+++ b/src/common/DataConversionFlowGraph.cpp
@@ -128,7 +128,7 @@ Result DataConversionFlowGraph::configure(AudioStream *sourceStream, AudioStream
                                                                   actualSourceFramesPerCallback);
                 break;
             default:
-                LOGE("%s() Unsupported source caller format = %s", __func__, oboe::convertToText(sourceFormat));
+                LOGE("%s() Unsupported source caller format = %d", __func__, static_cast<int>(sourceFormat));
                 return Result::ErrorIllegalArgument;
         }
         mSourceCaller->setStream(sourceStream);
@@ -150,7 +150,7 @@ Result DataConversionFlowGraph::configure(AudioStream *sourceStream, AudioStream
                 mSource = std::make_unique<SourceI32>(sourceChannelCount);
                 break;
             default:
-                LOGE("%s() Unsupported source format = %s", __func__, oboe::convertToText(sourceFormat));
+                LOGE("%s() Unsupported source format = %d", __func__, static_cast<int>(sourceFormat));
                 return Result::ErrorIllegalArgument;
         }
         if (isInput) {
@@ -226,7 +226,7 @@ Result DataConversionFlowGraph::configure(AudioStream *sourceStream, AudioStream
             mSink = std::make_unique<SinkI32>(sinkChannelCount);
             break;
         default:
-            LOGE("%s() Unsupported sink format = %s", __func__, oboe::convertToText(sinkFormat));
+            LOGE("%s() Unsupported sink format = %d", __func__, static_cast<int>(sinkFormat));
             return Result::ErrorIllegalArgument;;
     }
     lastOutput->connect(&mSink->input);

--- a/src/common/Utilities.cpp
+++ b/src/common/Utilities.cpp
@@ -278,6 +278,20 @@ const char *convertToText<ChannelCount>(ChannelCount channelCount) {
     }
 }
 
+template<>
+const char *convertToText<SampleRateConversionQuality>(SampleRateConversionQuality sampleRateConversionQuality) {
+
+    switch (sampleRateConversionQuality) {
+        case SampleRateConversionQuality::None:     return "None";
+        case SampleRateConversionQuality::Fastest:  return "Fastest";
+        case SampleRateConversionQuality::Low:      return "Low";
+        case SampleRateConversionQuality::Medium:   return "Medium";
+        case SampleRateConversionQuality::High:     return "High";
+        case SampleRateConversionQuality::Best:     return "Best";
+        default:                                    return "Unrecognized sample rate conversion quality";
+    }
+}
+
 std::string getPropertyString(const char * name) {
     std::string result;
 #ifdef __ANDROID__

--- a/src/opensles/AudioOutputStreamOpenSLES.cpp
+++ b/src/opensles/AudioOutputStreamOpenSLES.cpp
@@ -372,7 +372,7 @@ Result AudioOutputStreamOpenSLES::requestFlush_l() {
     } else {
         SLresult slResult = (*mSimpleBufferQueueInterface)->Clear(mSimpleBufferQueueInterface);
         if (slResult != SL_RESULT_SUCCESS){
-            LOGW("Failed to clear buffer queue. OpenSLES error: %d", result);
+            LOGW("Failed to clear buffer queue. OpenSLES error: %s", getSLErrStr(slResult));
             result = Result::ErrorInternal;
         }
     }

